_posts/2017-01-30-docker-frontend-tools.md

```diff
@@ -0,0 +1,146 @@
+
+We plan to post a more in-depth series of posts covering Docker for local development and CI environments in the near future, so be on the look out for that!
+
+We're also fans of using style guides and atomic design to drive front-end development. So when we heard of the [Pattern Lab Starter Drupal theme](https://github.com/phase2/pattern-lab-starter) which leverages [Pattern Lab](http://patternlab.io/) (a tool for building pattern-driven user interfaces using atomic design), we were very excited to try it out.
+
+## Building the Docker image
+
+To try out the Pattern Lab Starter theme we started with a fresh local Drupal 8 installation, and then quickly spun up our local Docker development environment [using Docker4Drupal](http://docs.docker4drupal.org/en/latest/#usage). We then copied the [Pattern Lab Starter code](https://github.com/phase2/pattern-lab-starter) to a new `custom/theme/patter_lab_starter` directory in our Drupal project.
+
+However, running the Phase 2 Pattern Lab Starter theme requires Node, npm, PHP, and Composer - the former for the node dependencies (Gulp, Bower, etc.) and the latter for running Pattern Lab.
+
+While we could install these dependencies locally, that defeats the purpose of using Docker. One of the great advantages of virtualization, be it Docker or a full VM, is that you don't have to rely on installing global dependencies on your local machine. One of the many benefits of this is that it ensures each team member is developing in the same environment.
+
+Unfortunately, while Docker4Drupal provides public images for many applications (such as Nginx, PHP, MariaDB, Mailhog, Redis, Apache Solr, and Varnish), it does not provide images for running the Pattern Lab Starter theme dependencies.
+
+One of the nice features of Docker though is that it is relatively easy to create a new image that builds upon other images. This is done via a `Dockerfile` which specifies the commands for creating the image.
+
+To build an image with our theme's dependencies we created a `Dockerfile` with the following contents:
+
+```text
+FROM node:7.1
+MAINTAINER Dan Murphy <dan@savaslabs.com>
+
+RUN apt-get update && \
+    apt-get install -y php5-dev  && \
+    curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer && \
+
+    # Directory required by Yeoman to run.
+    mkdir -p /root/.config/configstore \
+
+    # Clean up.
+    apt-get clean && \
+    rm -rf \
+			/root/.composer \
+			/tmp/* \
+			/usr/include/php \
+			/usr/lib/php5/build \
+			/var/lib/apt/lists/*
+
+# Permissions required by Yeoman to run: https://github.com/keystonejs/keystone/issues/1566#issuecomment-217736880
+RUN chmod g+rwx /root /root/.config /root/.config/configstore
+
+EXPOSE 3001 3050
+```
+
+The commands in this `Dockerfile`:
+
+- Set [the official Node 7 image](https://hub.docker.com/_/node/) as the base image. This base image includes Node and npm.
```

Comment:
Hi @kostajh - I'd prefer leaving it as just npm as they do on their site and [in their documentation](https://docs.npmjs.com/), especially since I'm referring to the application and not a specific command. Later in the post I do reference the `npm start` and `npm install` and use the back ticks. I also refer to Composer, Gulp and Bower without back ticks even though they are also command line tools.

I did update the text that introduces these applications to `However, running the Phase 2 Pattern Lab Starter theme requires Node.js, the node package manager npm, PHP, and the PHP dependency manager Composer.` and included links to their project homepages.